spacy/tokens/doc.pyx

```diff
@@ -1362,7 +1347,7 @@ cdef int _set_lr_kids_and_edges(TokenC* tokens, int length, int loop_count) exce
     for i in range(length):
         child = &tokens[i]
         head = &tokens[i + child.head]
-        if child < head and loop_count == 0:
+        if loop_count == 0 and child < head:
```

Comment:
Good idea reversing these conditions.

Reply:
The `loop_count` code is kind of dumb, but I tried splitting into two functions instead, but it didn't seem like an improvement.